benchmark/utils/general.hpp

```diff
@@ -290,30 +290,35 @@ void backup_results(rapidjson::Document& results)
 
 
 // executor mapping
-const std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>>
+const std::map<std::string, std::function<std::shared_ptr<gko::Executor>(bool)>>
```

Comment:
I think it will require the `timer.hpp` include the `general.hpp` or include `general.hpp` earlier than `timer.hpp`.
it only uses one input so I prefer not introducing another headers hierarchy.